planner/core/prepare_test.go

```diff
@@ -676,6 +676,39 @@ func (s *testPlanSerialSuite) TestPlanCacheUnionScan(c *C) {
 	c.Check(cnt, Equals, float64(6))
 }
 
+func (s *testPlanSerialSuite) TestPlanCacheHitInfo(c *C) {
+	defer testleak.AfterTest(c)()
+	store, dom, err := newStoreWithBootstrap()
+	c.Assert(err, IsNil)
+	tk := testkit.NewTestKit(c, store)
+	orgEnable := core.PreparedPlanCacheEnabled()
+	defer func() {
+		dom.Close()
+		store.Close()
+		core.SetPreparedPlanCache(orgEnable)
+	}()
+	core.SetPreparedPlanCache(true)
+
+	tk.Se, err = session.CreateSession4TestWithOpt(store, &session.Opt{
+		PreparedPlanCache: kvcache.NewSimpleLRUCache(100, 0.1, math.MaxUint64),
+	})
+	c.Assert(err, IsNil)
+
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(id int)")
+	tk.MustExec("insert into t values (1),(2),(3),(4)")
+	tk.MustExec("prepare stmt from 'select * from t where id=?'")
+	tk.MustExec("set @doma = 1")
+	tk.MustQuery(`show session variables like "last_statement_found_in_plan_cache"`).Check(testkit.Rows(
+		"last_statement_found_in_plan_cache 0",
+	))
+	tk.MustQuery("execute stmt using @doma").Check(testkit.Rows("1"))
+	tk.MustQuery(`show session variables like "last_statement_found_in_plan_cache"`).Check(testkit.Rows(
+		"last_statement_found_in_plan_cache 1",
+	))
```

Comment:
please execute another SQL that not found in the plan cache, and check the value of `@@last_statement_found_in_plan_cache` equals to `1`